pkg/agent/clientset.go

```diff
@@ -187,10 +205,16 @@ func (cs *ClientSet) sync() {
 			} else {
 				klog.ErrorS(err, "cannot connect once")
 				duration = backoff.Step()
+				if cs.onConnect != nil {
+					cs.SetConnectionStatus(false)
+				}
```

Comment:
In the multiple-control plane nodes case, won't this set to False incorrectly in some cases?

What about: setting status (somewhere in the for loop, unconditionally), based on `cs.ClientsCount() > 0`?

Reply:
https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/485#discussion_r1213143798